packages/tenable_ot_security/data_stream/system_log/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,119 @@
+
+##################### Scripts for processing input #####################
+  - script:
+      description: Drops null/empty values recursively.
+      lang: painless
+      source: |
+        boolean dropEmptyFields(Object object) {
+          if (object == null || object == "") {
+            return true;
+          } else if (object instanceof Map) {
+            ((Map) object).values().removeIf(value -> dropEmptyFields(value));
+            return (((Map) object).size() == 0);
+          } else if (object instanceof List) {
+            ((List) object).removeIf(value -> dropEmptyFields(value));
+            return (((List) object).length == 0);
+          }
+          return false;
+        }
+        dropEmptyFields(ctx);
+        
+  - script:
+      description: Convert Additional Info keys to snake case.
+      tag: additional-info-keys-to-snake-case
+      lang: painless
+      source: |
+        Map keysToSnakeCase(Map m) {
+          def regex = /_?([a-z])([A-Z]+)/;
+          def snakeCaseMap = [:];
+          for (entry in m.entrySet()) {
+            def k = entry.getKey();
+            def v = entry.getValue();
+            if (v instanceof Map) {
+              v = keysToSnakeCase(v);
+            } else if (v instanceof List) {
+              for (int i = 0; i < v.size(); i++) {
+                def item = v.get(i);
+                if (item instanceof Map) {
+                  v.set(i, keysToSnakeCase(item));
+                }
+              }
+            }
+            k = regex.matcher(k).replaceAll('$1_$2').toLowerCase();
+            snakeCaseMap.put(k, v);
+          }
+          return snakeCaseMap;
+        }
+        if (ctx.tenable_ot_security.system_log != null) {
+          ctx.tenable_ot_security.system_log = keysToSnakeCase(ctx.tenable_ot_security.system_log);
+        }
+##################### ECS Fields #####################
+  - set:
+      field: event.kind
+      value: event
+
+  - set:
+      field: ecs.version
+      value: "8.11.0"
+
+  - append:
+      field: event.category
+      value: iam
+
+  - set:
+      field: event.module
+      value: tenable_ot_security
+
+  - set:
+      field: event.type
+      value: ["info"]
+
+  - set:
+      field: event.dataset
+      value: tenable_ot_security.system_log
+
+  - set:
+     field: "message"
+     copy_from: tenable_ot_security.system_log.message
+     ignore_empty_value: true
+
+  - set:
+     field: "@timestamp"
+     copy_from: tenable_ot_security.system_log.time_stamp
+     ignore_empty_value: true
+
+  - set:
+     field: "user.name"
+     value: "{{{tenable_ot_security.system_log.user_name}}}"
+     ignore_empty_value: true
+     
+  - append:
+      field: "related.user"
+      value: "{{{tenable_ot_security.system_log.user_name}}}"
+      allow_duplicates: false
```

The quotes around field values (except for `"@timestamp"`) are redundant and can be removed as follows.
However, these processors should probably be renames in order to avoid redundant vendor prefixed fields, as mentioned elsewhere.
```suggestion
  - set:
     field: message
     copy_from: tenable_ot_security.system_log.message
     ignore_empty_value: true

  - set:
     field: "@timestamp"
     copy_from: tenable_ot_security.system_log.time_stamp
     ignore_empty_value: true

  - set:
     field: user.name
     value: "{{{tenable_ot_security.system_log.user_name}}}"
     ignore_empty_value: true
     
  - append:
      field: related.user
      value: "{{{tenable_ot_security.system_log.user_name}}}"
      allow_duplicates: false
```